runtime/scripts/part.js

```diff
@@ -1238,6 +1251,7 @@ if(res) { \
                 }
             }
         }
+        this.events.trigger('submit');
```

Comment:
This would definitely benefit from triggering an event at either end of the function.